### DISCUSSION TITLE
Profile variables as Claims for holding extra user data

### DISCUSSION THREAD

**Discussion by markdav-is** [oqtane/oqtane.framework#2810] [Q&A]
We have a need to identify our users beyond role membership.  ASP.Net Claims is the logical choice to hold this extra data.  I was thinking about Oqtane user profiles and the settings sub-system as a claims-like way to add extra user data (for example if a user is a teacher and the names of the affiliated school).  

I'm considering using the profile settings to store these claims and then extending the build of the claimsPrincipal to include them at login, while roles are added. 

My question is:  Is this a solid approach?  Am I missing something that might already be included in Oqtane that would serve this purpose better?

**Comment by sbwalker** [comment]
@markdav-is this is already implemented:

![image](https://github.com/oqtane/oqtane.framework/assets/4840590/8ba9c0b2-bf47-446a-ba24-2f8ac8f68a7b)

**Reply by markdav-is** [reply]
Thanks for answering @sbwalker.  I'm implementing this this week.  We are not using OIDC but I'm checking to see how we can pull the oqtane profile information in the controllers to use them for filtering data based on claims.
